File: 백준/Silver/4948. 베르트랑 공준/베르트랑 공준.java
import java.io.*;
import java.util.Arrays;

public class Main{
    public static void main(String[] args) throws IOException{

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int MAX =  123456 * 2;
        int[] primeCount = new int[MAX+1];
        Boolean[] isPrime = new Boolean[MAX+1];
        Arrays.fill(isPrime, true);

        for (int i=2; i<(int)Math.sqrt(MAX+1); i++) {

            // i이 현재까지 True 인 경우 확인
            if (isPrime[i]) {
                int j = 2;

                while(i * j <= MAX) {
                    isPrime[i*j] = false;
                    j += 1;
                }
            }
        }

        for (int i=1; i<MAX+1; i++) {
            if (isPrime[i])
                primeCount[i] = primeCount[i-1] + 1;
            else
                primeCount[i] = primeCount[i-1];
        }

        while (true) {
            int n = Integer.parseInt(br.readLine());

            if (n == 0)
                break;

            bw.write(primeCount[2*n] - primeCount[n]+ "\n");
        }

        bw.flush();

    }
}

File: 백준/Silver/4948. 베르트랑 공준/베르트랑 공준.py
import math

if __name__ == '__main__':
    # 100,000 = 10,000,000,000 5

    MAX = 123456 * 2
    prime_count = [0 for _ in range(MAX+1)]
    prime = [True for i in range(MAX + 1)]

    prime[1] = False

    for i in range(2, int(math.sqrt(MAX)) + 1):
        if prime[i]:
            j = 2
        while i * j <= MAX:
            prime[i * j] = False
            j += 1

    for i in range(1, MAX+1):
        if prime[i]:
            prime_count[i] = prime_count[i-1] + 1
        else:
            prime_count[i] = prime_count[i-1]

    while 1:
        n = int(input())

        if n == 0:
            break

        print(prime_count[2 * n] - prime_count[n])